src/components/stateful-enroll/data/service.js

```diff
@@ -1,59 +1,47 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import { camelCaseObject } from '@edx/frontend-platform/utils';
 import { getConfig } from '@edx/frontend-platform/config';
-import { v4 as uuidv4 } from 'uuid';
 
 /**
  * Makes an API request to retrieve the most recent payload for the
  * specified transaction UUID.
  *
- * Note: this arg may change to be given the actual API url to call
- * rather than just the `transactionUUID` (TBD during API integration).
- *
  * @param {object} args
- * @param {string} args.contentKey
  * @param {string} args.transactionUUID
  * @returns The payload for the specified transaction.
  */
 /* istanbul ignore next */
-export const retrieveTransactionStatus = async ({ contentKey, transactionUUID }) => {
-  const mockCommittedTransaction = {
-    uuid: transactionUUID,
-    state: 'committed',
-    courseware_url: `${getConfig().LEARNING_BASE_URL}/course/${contentKey}/home`,
-  };
-  const url = 'https://httpbin.org/post';
-  const response = await fetch(url, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(mockCommittedTransaction),
-  });
-  const result = await response.json();
-  return camelCaseObject(result.json);
+export const retrieveTransactionStatus = async ({ transactionUUID }) => {
+  const config = getConfig();
+
+  const url = `${config.ENTERPRISE_SUBSIDY_BASE_URL}/api/v1/transactions/${transactionUUID}/`;
```

Comment:
Similar to how `policyRedemptionUrl` is passed into `submitRedemptionRequest`, the intent of the API design is that the response from the API call to enterprise-access's `redeem` should be returning a transaction status API url to use rather than us needing to create the API url here ourselves, e.g. `const retrieveTransactionStatus = async ({ transactionStatusUrl }) => {`.

Does not block this PR, but I filed this ticket to follow-up when it makes for us to pick up: https://2u-internal.atlassian.net/browse/ENT-7183

Reply:
Thanks for filing the ticket!